pkg/kncloudevents/message_sender.go

```diff
@@ -46,41 +39,19 @@ var noRetries = RetryConfig{
 	},
 }
 
-// ConnectionArgs allow to configure connection parameters to the underlying
-// HTTP Client transport.
-type ConnectionArgs struct {
-	// MaxIdleConns refers to the max idle connections, as in net/http/transport.
-	MaxIdleConns int
-	// MaxIdleConnsPerHost refers to the max idle connections per host, as in net/http/transport.
-	MaxIdleConnsPerHost int
-}
-
-func (ca *ConnectionArgs) ConfigureTransport(transport *nethttp.Transport) {
-	if ca == nil {
-		return
-	}
-	transport.MaxIdleConns = ca.MaxIdleConns
-	transport.MaxIdleConnsPerHost = ca.MaxIdleConnsPerHost
-}
-
 type HTTPMessageSender struct {
 	Client *nethttp.Client
 	Target string
 }
 
-func NewHTTPMessageSender(connectionArgs *ConnectionArgs, target string) (*HTTPMessageSender, error) {
-	// Add connection options to the default transport.
-	var base = nethttp.DefaultTransport.(*nethttp.Transport).Clone()
-	connectionArgs.ConfigureTransport(base)
-	// Add output tracing.
-	client := &nethttp.Client{
-		Transport: &ochttp.Transport{
-			Base:        base,
-			Propagation: tracecontextb3.TraceContextEgress,
-		},
-	}
+// Deprecated: Don't use this anymore, now it has the same effect of NewHTTPMessageSenderWithTarget
+// If you need to modify the connection args, use ConfigureConnectionArgs sparingly.
+func NewHTTPMessageSender(ca *ConnectionArgs, target string) (*HTTPMessageSender, error) {
+	return NewHTTPMessageSenderWithTarget(target)
+}
 
-	return &HTTPMessageSender{Client: client, Target: target}, nil
+func NewHTTPMessageSenderWithTarget(target string) (*HTTPMessageSender, error) {
+	return &HTTPMessageSender{Client: getClient(), Target: target}, nil
```

Comment:
Just an observation/thought, probably silly, but if we pass in here interface to how to get the client, we don't need the global resync at the cost of the doing the Read lock on the mutex guarding the client. This might make it easier to write tests also? Curious on what the overhead for a read mutex would be.

Reply:
I completely disagree with this idea :smile: Putting a read lock, shared between all dispatchers in the instance, on the hot path, doesn't sound at all like a good idea IMO. 

To be honest, I think the configuration of these http params shouldn't be allowed at all during the execution of the dispatcher. It should be a single read when the process is started and that's it. These are params that are usually configured only at setup time, when you "setup" the capacity of your deployments: you don't change it on fly.

If you do, then you expect in any case a sort of "downtime"... So what I did in this PR is that I've done the bare minimum to fix the issue, I would love to open later a discussions on these params and why they should be configurable only at startup (and not like watches)

Reply:
Yeah, just jotted it down :) I figured you'd be a fan :) 
I'm curious also on how often these in practice need to change.

Reply:
I guess never, hence my idea of removing that watch

Reply:
If that complexity is not needed in the first place, removing it in another PR sounds like an excellent idea to me. Does anyone know if there is/was a use-case in Eventing for reconfiguring the transport on the fly?

Reply:
https://github.com/knative/eventing/issues/4469